mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/NextStepBanner.kt

```diff
@@ -0,0 +1,114 @@
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.mobile.ui.PawniesColors
+import ch.epfl.sdp.mobile.ui.PawniesIcons
+import ch.epfl.sdp.mobile.ui.TournamentsNextStep
+
+/**
+ * A green variation of [NextStepBanner].
+ *
+ * @see NextStepBanner
+ */
+@Composable
+fun GreenNextStepBanner(
+    title: String,
+    message: String,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+  NextStepBanner(
+      title = title,
+      message = message,
+      onClick = onClick,
+      titleColor = PawniesColors.Green800,
+      messageColor = PawniesColors.Green500,
+      backgroundColor = PawniesColors.Green100.copy(alpha = 0.4f),
+      borderColor = PawniesColors.Green200,
+      modifier = modifier,
+  )
+}
+
+/**
+ * An orange variation of [NextStepBanner].
+ *
+ * @see NextStepBanner
+ */
+@Composable
+fun OrangeNextStepBanner(
+    title: String,
+    message: String,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+  NextStepBanner(
+      title = title,
+      message = message,
+      onClick = onClick,
+      titleColor = PawniesColors.Orange500,
+      messageColor = PawniesColors.Orange500,
+      backgroundColor = PawniesColors.Orange200.copy(alpha = 0.2f),
+      borderColor = PawniesColors.Orange500,
+      modifier = modifier,
+  )
+}
+
+/**
+ * A banner which indicates a next step of a round.
+ *
+ * @param title the title of the banner.
+ * @param message the message of the banner.
+ * @param onClick the listener called when the banner is clicked.
+ * @param titleColor the color of the title.
+ * @param messageColor the color of the message.
+ * @param backgroundColor the color of the background.
+ * @param borderColor the color of the border.
+ * @param modifier the [Modifier] for this composable.
+ */
+@Composable
+private fun NextStepBanner(
+    title: String,
+    message: String,
+    onClick: () -> Unit,
+    titleColor: Color,
+    messageColor: Color,
+    backgroundColor: Color,
+    borderColor: Color,
+    modifier: Modifier = Modifier,
+) {
+  Row(
+      modifier =
+          modifier
+              .fillMaxWidth()
+              .border(2.dp, borderColor, RoundedCornerShape(16.dp))
+              .clip(RoundedCornerShape(16.dp))
+              .clickable { onClick() }
+              .background(backgroundColor)
+              .padding(16.dp),
+      horizontalArrangement = Arrangement.SpaceBetween,
+      verticalAlignment = Alignment.CenterVertically,
+  ) {
+    Column(Modifier.weight(1f)) {
+      Text(text = title, style = MaterialTheme.typography.subtitle1, color = titleColor)
+      Text(text = message, style = MaterialTheme.typography.body1, color = messageColor)
+    }
+    CompositionLocalProvider(LocalContentColor provides titleColor) {
```

Comment:
Question: What does this LocalProvider?

Reply:
It makes sure that the icon's color matches the title color.